client/my-sites/customer-home/locations/primary/index.jsx

```diff
@@ -48,23 +55,29 @@ const cardComponents = {
 	[ NOTICE_CELEBRATE_SITE_SETUP_COMPLETE ]: CelebrateSiteSetupComplete,
 };
 
-const Primary = ( { cards } ) => {
+const Primary = ( { cards, renderCard } ) => {
 	if ( ! cards || ! cards.length ) {
 		return null;
 	}
 
-	return (
-		<>
-			{ cards.map(
-				( card, index ) =>
-					cardComponents[ card ] &&
-					React.createElement( cardComponents[ card ], {
-						key: index,
-						isIos: card === 'home-task-go-mobile-ios' ? true : null,
-					} )
-			) }
-		</>
-	);
+	return <>{ cards.map( renderCard ) }</>;
 };
 
-export default withPerformanceTrackerStop( Primary );
+export default connect( null, ( dispatch ) => ( {
+	renderCard: ( card ) =>
```

Comment:
Alternatively if we end up keeping this: with this many lines, it might be more readable to use braces and return a value in this function